Bindings/Java/Matlab/Hopper_Device/RunHopperWithDevice_answers.m

```diff
@@ -61,9 +61,9 @@
 anchorA = WeldJoint.safeDownCast(device.updComponent('anchorA'));
 anchorB = WeldJoint.safeDownCast(device.updComponent('anchorB'));
 thighAttach = PhysicalFrame.safeDownCast(...
-        hopper.getComponent('thigh/deviceAttach'));
+        hopper.getComponent('bodyset/thigh/deviceAttach'));
```

Comment:
In a model created programmatically, is it "required" now to have bodyset in Pathname? Does it depend on how the Body is added to the model?

Reply:
Yes if you add a body to the Model's `BodySet` you have to include *bodyset* in the path. to retrieve it. That was the point of the change to distinguish components added to different Sets by recognizing the Set as part of the pathname.

Reply:
This should be documented somewhere since it is not obvious that addBody(body) and addComponent(body) and addModelComponent(body) would result in 3 different layouts with different pathnames.

Reply:
`Model` has documentation in the section for adding components to the model. I have added some addition info to indicate the pathnames are different.
```cpp
    /** @name Adding components to the Model
     * Model takes ownership of the ModelComponents and adds them to specialized
     * (typed) Sets within the model. These sets can be useful for uniquely
     * identifying components that share the same name but are of different types
     * since they will live in different sets. For example, using addBody(toesBody)
     * and addJoint(toesJoint) will have unique paths "/bodyset/toes" and 
     * "/jointset/toes", respectively even if they share the same name, "toes".
     * Model will maintain Components added using these methods in separate %Sets
     * of the corresponding type and they will serialize as part of type specific
     * %Sets.
     * @note these are legacy methods and remain as a convenience alternative to
     * using Component::addComponent().
     * In contrast, components added using addComponent() are not stored in the
     * model's %Sets but live in a flat components list (which is also serialized).
     * Component provides access via getComponentList<%SpecificType> or 
     * getComponent<%SpecificType> to get any subcomponent, including those that
     * are contained in Model's %Sets. In this case, either a Body or a Joint has
     * the pathname "/toes" but both cannot share the same name and will throw
     * SubcomponentsWithDuplicateName.
     * Future versions of OpenSim are likely to deprecate the use of Sets and
     * these methods, because they cannot support new types without modifying the
     * API (for more add####() methods), whereas getComponentList<%SpecificType>()
     * and getComponent<%SpecificType> are more general: they do not have these
     * limitations and are applicable for any Component not just Model.
     */
```